src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallerFactory.cs

```diff
@@ -56,5 +61,27 @@ public static InstallType GetWorkloadInstallType(SdkFeatureBand sdkFeatureBand,
 
             return InstallType.FileBased;
         }
+
+        private static bool CanWriteToDotnetRoot(string dotnetDir = null)
+        {
+            dotnetDir = dotnetDir ?? Path.GetDirectoryName(Environment.ProcessPath);
+            try
+            {
+                var testPath = Path.Combine(dotnetDir, "metadata", Path.GetRandomFileName());
+                if (Directory.Exists(Path.GetDirectoryName(testPath)))
+                {
+                    using (FileStream fs = File.Create(testPath, 1, FileOptions.DeleteOnClose)) { }
+                }
+                else
+                {
+                    Directory.CreateDirectory(Path.GetDirectoryName(testPath));
+                }
+                return true;
+            }
+            catch
```

Comment:
Definitely catch unauthorized exception to return false. Any other exception doesn't necessarily mean you don't have permission, e.g. you could get an IOException if something else failed in the OS that's not permission related and you can't be sure it will definitely fail again, unless it's a permission issue